lua/mini/misc.lua

```diff
@@ -409,6 +409,91 @@ MiniMisc.zoom = function(buf_id, config)
   end
 end
 
+--- Restore the cursor on file reopen
+---
+--- What it does:
+--- - When reopening a file, this will make sure the cursor is placed back to
+---   the position where you left before. This implements `:help restore-cursor`
+---   in a nicer way.
+---
+--- If you want to lazy load this use the |BufReadPre| event.
+---
+---@param opts table|nil Options for |MiniMisc.restore_cursor|. Possible fields:
+---   - <ignore_buftype> - (table) Define buffer types to be ignored.
+---     Default: { "quickfix", "nofile", "help" }
+---   - <ignore_filetype> - (table) Define file types to be ignored.
+---     Default: { "gitcommit", "gitrebase" }
+---   - <center> - (boolean) Center the window after we restored the cursor.
+---     Default: true
+---
+---@usage >
+---   require('mini.misc').setup_restore_cursor()
+MiniMisc.setup_restore_cursor = function(opts)
+  opts = opts or {}
+  opts.ignore_buftype = opts.ignore_buftype or { "quickfix", "nofile", "help" }
+  opts.ignore_filetype = opts.ignore_filetype or { "gitcommit", "gitrebase" }
+  opts.center = opts.center or true
+
+  if not H.is_array_of(opts.ignore_buftype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_buftype` should be an array of strings.')
+  end
+
+  if not H.is_array_of(opts.ignore_filetype, H.is_string) then
+    H.error('In `setup_restore_cursor()` `opts.ignore_filetype` should be an array of strings.')
+  end
+
+  if not H.is_boolean(opts.center) then
+    H.error('In `setup_restore_cursor()` `opts.center` should be a boolean.')
+  end
+
+  local au_command = string.format(
+    [[augroup MiniMiscRestoreCursor
+      au!
+      au BufReadPre * au FileType <buffer> ++once lua require('mini.misc').restore_cursor(%s)
+    augroup END]],
+    vim.inspect(opts, { newline = ' ', indent = '' }))
+  vim.api.nvim_exec(au_command, false)
+end
+
+--- Restore cursor when reopening a file
+MiniMisc.restore_cursor = function(opts)
+  -- Check if the buffer should be ignored
+  if vim.tbl_contains(opts.ignore_buftype, vim.bo.buftype) then
+    return
+  end
+
+  -- Check if the filetype should be ignored
+  if vim.tbl_contains(opts.ignore_filetype, vim.bo.filetype) then
+      return
+  end
+
+  -- If a line has already been specified on the command line, we are done
+  --   nvim file +num
+  if vim.fn.line('.') > 1 then
+      return
+  end
```

Comment:
I am not entirely sure, but is there a test for this early return?

Reply:
How do I restart the child with a file and line specified?

Reply:
Try `child.restart({ '+2', '--', test_file_cursor })`

Reply:
I get:

```
Error executing lua: [string "<nvim>"]:1: attempt to index global 'MiniMisc' (a nil value)
```

for `restore_cursor()`.

Reply:
Yes. It restarts child process with putting cursor on specific line. You also need to set up 'mini.misc'. Use `load_module()`.

Reply:
`load_module()` and specifying the config is needed to find the module. Works now.